src/KurrentDB.Core/Services/Transport/Http/HttpContextExtensions.cs

```diff
@@ -11,4 +12,15 @@ public static bool IsUnixSocketConnection(this HttpContext ctx) {
 		var connectionItemsFeature = ctx.Features.Get<IConnectionItemsFeature>();
 		return connectionItemsFeature is not null && connectionItemsFeature.Items.ContainsKey(UnixSocketConnectionMiddleware.UnixSocketConnectionKey);
 	}
+
+	[CanBeNull]
+	public static DuckDBConnectionPool GetDuckDbConnectionPool(this HttpContext httpContext) {
```

Comment:
It should never be null.

```csharp
public static class HttpContextExtensions {
    extension(HttpContext ctx) {
        public DuckDBConnectionPool GetDuckDbConnectionPool() =>
            ctx.Features.Get<IConnectionItemsFeature>()?.Items.TryGetValue(nameof(DuckDBConnectionPool), out var item) != true ||
            item is not DuckDBConnectionPool pool
                ? throw new UnreachableException("No DuckDBConnectionPool available for this connection")
                : pool;
    }
}
```

Reply:
:+1: i've made this method specifically for getting the connection scoped pool with no fallback (because whoever is asking for it knows what they need), and ive made it throw if its not available